src/scripts/main.js

```diff
@@ -1,3 +1,242 @@
 'use strict';
 
-// write your code here
+const score = document.querySelector('.game-score');
+const cell = document.querySelectorAll('.field-cell');
+const button = document.querySelector('.button');
+
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+const messageStart = document.querySelector('.message-start');
+
+const rows = 4;
+const columns = 4;
+
+button.addEventListener('click', () => {
+  if (button.textContent === 'Start') {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.textContent = 'Restart';
+  } else {
+    button.classList.remove('restart');
+    button.classList.add('start');
+    button.textContent = 'Start';
+  }
+
+  score.textContent = '0';
+  messageWin.classList.add('hidden');
+  messageLose.classList.add('hidden');
+  messageStart.classList.add('hidden');
+
+  beginGame();
+});
+
+let board;
+
+function beginGame() {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  let i = 0;
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      cell[i].id = `${r}${c}`;
+
+      const num = board[r][c];
+
+      updateTile(cell[i], num);
+      i++;
+    }
+  }
+
+  setRandom();
+  setRandom();
+  document.addEventListener('keyup', handler);
+}
+
+function hasEmptyTile() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function setRandom() {
+  if (!hasEmptyTile()) {
+    return;
+  }
+
+  let findEmptyTile = false;
+
+  while (!findEmptyTile) {
+    const r = Math.floor(Math.random() * rows);
+    const c = Math.floor(Math.random() * columns);
+
+    if (board[r][c] === 0) {
+      board[r][c] = Math.random() > 0.9 ? 4 : 2;
+
+      const tile = document.getElementById(`${r}${c}`);
+
+      tile.innerText = `${board[r][c]}`;
+      tile.classList.add(`field-cell--${board[r][c]}`);
+      findEmptyTile = true;
+    }
+  }
+}
+
+function updateTile(tile, num) {
+  tile.innerText = '';
+  tile.classList.value = '';
+  tile.classList.add('field-cell');
+
+  if (num > 0) {
+    tile.innerText = num;
+    tile.classList.add(`field-cell--${num}`);
+  }
+}
+
+const handler = (e) => {
+  switch (e.code) {
+    case 'ArrowLeft': {
+      slideLeftRight(true);
+      setTimeout(setRandom, 500);
```

Comment:
remove timeout - when the user clicks fast new tiles appears too late (and that might be the reason for the problem with a too-early message about the user loss)